lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1705,16 +1705,7 @@ getAccessorPropertyType(const clang::FunctionDecl *accessor, bool isSetter,
 /// of this class as Swift generic type params.
 static bool
 shouldSuppressGenericParamsImport(const clang::ObjCInterfaceDecl *decl) {
-  while (decl) {
-    StringRef name = decl->getName();
-    if (name == "NSArray" || name == "NSDictionary" || name == "NSSet" ||
-        name == "NSOrderedSet" || name == "NSEnumerator" ||
-        name == "NSMeasurement") {
-      return true;
-    }
-    decl = decl->getSuperClass();
-  }
-  return false;
+  return decl->hasAttr<clang::SwiftImportAsNonGenericAttr>();
```

Comment:
It does not get inherited, which I think is the desired behavior going forward, but yeah, maybe it's a good idea for Swift 3 compatibility mode to walk the superclass chain.